src/storage/store/chunked_node_group.cpp

```diff
@@ -203,11 +203,17 @@ static ZoneMapCheckResult getZoneMapResult(const TableScanState& scanState,
             if (columnID == INVALID_COLUMN_ID || columnID == ROW_IDX_COLUMN_ID) {
                 continue;
             }
-            KU_ASSERT(i < scanState.columnPredicateSets.size());
-            const auto columnZoneMapResult = scanState.columnPredicateSets[i].checkZoneMap(
-                chunks[columnID]->getData().getMergedColumnChunkStats());
-            if (columnZoneMapResult == common::ZoneMapCheckResult::SKIP_SCAN) {
-                return common::ZoneMapCheckResult::SKIP_SCAN;
+
+            const bool columnHasStorageValueType =
+                TypeUtils::visit(chunks[columnID]->getDataType().getPhysicalType(),
+                    []<typename T>(T) { return StorageValueType<T>; });
```

Comment:
Found some weirdness going on in the RDF tests where we can cast an RDF value (physical type struct) to an integer. I'm not sure if we can do that with normal structs but here it only checks the zone map if the column has a storage value type just in case.

Reply:
I would remove those RDF test since we are removing RDF anyways and make this an assertion.

Reply:
added assertion that we only skip scan if `columnHasStorageValueType` is true